pkg/k8s/watchers/watcher.go

```diff
@@ -454,7 +444,7 @@ var ciliumResourceToGroupMapping = map[string]watcherInfo{
 
 // resourceGroups are all of the core Kubernetes and Cilium resource groups
 // which the Cilium agent watches to implement CNI functionality.
-func (k *K8sWatcher) resourceGroups() (beforeNodeInitGroups, afterNodeInitGroups []string) {
+func (k *K8sWatcher) resourceGroups() []string {
```

Comment:
Was this only used for CEC and CCEC resources?

Reply:
Yes. The reason is that the local node (with its Ingress IPs) needs to be fully initialized before starting to watch CEC/CCECs ([initial commit](https://github.com/cilium/cilium/commit/a140e145e55)).

We still ensure this by [depending on the](https://github.com/cilium/cilium/pull/30298/files#diff-33945f18eb5964fb8de0b8f6d88e59d1520bb94c24620634b7e4bfe0763d583c) `LocalNodeStore` and rely on the fact hat its Hive Cell gets initialized before the CEC cell (and therefore the global variable with the Local Node Ingress IPs gets initialized before we [start accessing it during reconciliation](https://github.com/cilium/cilium/blob/main/pkg/envoy/xds_server.go#L826-L841) (`node.GetIngressIPv4`)).

Part II (https://github.com/cilium/cilium/pull/30290) will improve this further by retrieving the Node' Ingress IP during cell initialization instead of accessing global functions. This way it should become more explicit and easier to understand.